src/markdown/footnotes.scss

```diff
@@ -0,0 +1,26 @@
+// stylelint-disable selector-max-type
+// stylelint-disable selector-no-qualifying-type
+.markdown-body .footnotes {
+  font-size: $h6-size;
+  color: var(--color-fg-muted);
+  border-top: $border;
+
+  ol {
+    padding-left: $spacer-3;
+    margin-bottom: $spacer-3;
+  }
+
+  p {
+    margin-top: $spacer-3;
+  }
+
+  a.data-footnote-backref g-emoji {
+    font-family: monospace;
+  }
+
+  :target {
+    border: $border-width $border-style var(--color-accent-emphasis);
```

Comment:
I'm noticing quite a jump in the footnote element when it receives the `:target` pseudo-class:

> ![gif](https://user-images.githubusercontent.com/11878752/134399922-70e00f95-c282-49c1-99e2-2448d583d7bb.gif)

- Can we make sure the border doesn't affect the position of the inner elements?
- Can we position the borders in such a way it that they don't touch the edges? See example below.
- Should we be specific about this `:target` rule? It's currently being applied to **any** element inside `.markdown-body .footnotes`, but maybe we should be more specific.

Example of surrounding border better respecting the spacing around:
> ![Screen Shot 2021-09-24 at 13 19 47](https://user-images.githubusercontent.com/293280/134734957-a0f83532-b6b8-405c-8d68-57121227b959.png)
> [Link to Figma exploration](https://www.figma.com/file/Nqpm7TZ4BcIekrdPxG8G4r/quick-markdown-footnote-support?node-id=101%3A8)

Reply:
Makes sense...I think this is going to be as close as I can get due to time constraints. I'll play around a little more on Monday and push it up. Thanks for your guidance and patience! I haven't done my own CSS in quite a bit of time. ✨ 
<img width="862" alt="Screen Shot 2021-09-24 at 6 14 45 PM" src="https://user-images.githubusercontent.com/11878752/134745103-630f7287-e2cf-4795-85e9-765c360541ff.png">


Eliminated bullet point positioning jump 
![footnotes_css](https://user-images.githubusercontent.com/11878752/134745110-92b68deb-56b4-4a1c-9655-5e05cefaf0b7.gif)

Reply:
This is a bit tricky because the list-item numbers are "outside" of the `<li>`. Tried using [`list-style-position: inside;`](https://developer.mozilla.org/en-US/docs/Web/CSS/list-style-position), but that caused problems with multi-line content like pagraphs.

This PR https://github.com/primer/css/pull/1624 uses a `::before` pseudo element to "draw" the `:target` style with negative positioning so it looks like the list items get wrapped:

![Screen Shot 2021-09-27 at 17 35 00](https://user-images.githubusercontent.com/378023/134874559-1cb33c94-996f-4529-8c21-0109839e19b3.png)

Reply:
Oh, I like it!!! Yeah, I was trying a similar technique in [this branch](https://github.com/github/github/compare/footnotes-styling-take-2). It did made the multi-line content strange, which I tried to get around by adding `margin-left` to any paragraph that wasn't the first child. 

I'm wondering if the right side should also have some positioning so that there's symmetry. Let me see. 

Thank you for your help 🙏🏼 . The `::before` pseudo element is interesting. I had tried to remove the jumpiness by adding a transparent border.

Reply:
Cherry-picked @simurai's commits. 🍒  Updated description ☝🏼 . 

Also added the darkened text when targeted.

I didn't change the spacing between paragraphs (that's in the mock). It seemed really tight to me and made it hard to read. 

Example:

<img width="1020" alt="Screen Shot 2021-09-27 at 9 19 05 AM" src="https://user-images.githubusercontent.com/11878752/134916396-b2e7ffe7-23e7-4bb7-afec-7755f77017ba.png">